src/rules/unique-dependencies.ts

```diff
@@ -19,43 +19,86 @@ const dependencyPropertyNames = new Set([
 	"peerDependencies",
 ]);
 
-// This is a list of groups that required uniqueness among all of them globally.
-const groupsRequiringGlobalUniqueness = new Set([
-	"dependencies",
-	"devDependencies",
-]);
-
-// This keep tracks of dependencies seen from above set.
 export const rule = createRule({
 	create(context) {
-		const globalUniqueDependenciesSeen = new Set();
+		// This is a list of groups that required uniqueness among all of them globally.
+		// Inside each object, groups defined the corresponding list of groups to check for.
+		// "remove" will determine in which dependency group should the duplicated dependency be removed.
+		// eg: if same dependency appears in "dependencies" and "deveDependencies" we keep that from "dependencies".
+		const groupsRequiringGlobalUniqueness = [
+			{
+				groups: ["dependencies", "devDependencies"],
+				remove: "devDependencies",
```

Comment:
The way this data structure works, a "group list" could only consist of two groups, right?  Any more than that, and you couldn't have a single string for `remove`.  Seems like a limitation to the api.  Not a deal breaker. Just calling it out that if we wanted to have global uniqueness across more than two groups, this logic would need to be refactored.